nuke_nodes.py

```
@@ -217,6 +225,35 @@ def update_blocked_node(node_id, comment=""):
     except subprocess.CalledProcessError as e:
         log_message("Failed to block node {}: {}".format(node_id, e))
 
+
+def is_node_blocked(node_id):
+    """Check if the node is currently blocked in the Asterisk database."""
+    if is_whitelisted(node_id):
+        return False
+    command = "{} {} -rx \"database get {} {}\"".format(SUDO, ASTERISK, DB_FAMILY, node_id)
+    try:
+        result = subprocess.run(
```

Ruff (0.14.13)

`subprocess` call with `shell=True` identified, security issue

(S602)